### PR TITLE
created component for auth

### PR DESCRIPTION
**Pull request by Las04ka**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -113,6 +119,7 @@
 	"cli": {
 		"schematicCollections": [
 			"@angular-eslint/schematics"
-		]
+    ],
+    "analytics": "377e5ac6-d9e6-4bec-89a5-b80db1342564"
```
Comment (AleaJactaEst) [inline angular.json:123]: do we need this?

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,17 +1,32 @@
 import { NgModule } from '@angular/core';
```
Comment (AleaJactaEst) [inline src/app/app-routing.module.ts:1]: Would prefer lazy loading for auth module

File: src/app/auth/components/auth/auth.component.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      align-items: center;
+    }
+
+    form {
+      display: flex;
+      flex-direction: column;
+    }
+
+    .links {
+      margin-top: 1vh;
```
Comment (AleaJactaEst) [inline src/app/auth/components/auth/auth.component.scss:22]: not a fun of 1vh - looks weird

File: src/app/auth/components/auth/auth.component.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.wrapper {
+  margin: 20vh auto;
+  width: 40%;
+  display: flex;
+  justify-content: center;
+
+  mat-card {
```
Comment (AleaJactaEst) [inline src/app/auth/components/auth/auth.component.scss:7]: lets try not to override like that, but instead use classes 

btw: i saw wrapper

File: src/app/auth/components/auth/auth.component.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    email: new FormControl(null, [Validators.required, Validators.email]),
+    password: new FormControl(null, [Validators.required]),
+    rememberMe: new FormControl(false),
+  });
+  constructor(
+    private router: Router,
+    private store: Store,
+    public activatedRoute: ActivatedRoute,
+  ) {
+    this.isRequesting = store.select(selectRequestingStatus);
```
Comment (AleaJactaEst) [inline src/app/auth/components/auth/auth.component.ts:29]: can be done on variable init

File: src/app/auth/components/auth/auth.component.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    this.isRegister = activatedRoute.snapshot.data['isRegister'];
+    this.title = activatedRoute.snapshot.data['title'];
+  }
+
+  onLogin(): void {
+    if (this.form.valid) {
```
Comment (AleaJactaEst) [inline src/app/auth/components/auth/auth.component.ts:35]: Kinda not a fun of method called onLogin that doing both login and signup
Lets create abstract class or 2 components

File: src/app/auth/components/auth/auth.component.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  styleUrls: ['./auth.component.scss'],
+})
+export class AuthComponent {
+  isRequesting: Observable<boolean>;
+  isRegister: boolean;
+  title: string;
+  form: FormGroup = new FormGroup({
+    email: new FormControl(null, [Validators.required, Validators.email]),
+    password: new FormControl(null, [Validators.required]),
+    rememberMe: new FormControl(false),
```
Comment (AleaJactaEst) [inline src/app/auth/components/auth/auth.component.ts:22]: Do we even use rememberMe?

File: src/app/auth/models/user.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export interface ILoginResponseData {
+  tokens: {
+    accessToken: string;
+  };
+  user: { id: number };
```
Comment (AleaJactaEst) [inline src/app/auth/models/user.ts:5]: Is that correct user model :) ?

File: src/app/auth/state/effects.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+@Injectable()
+export class AuthEffects {
+  constructor(
+    private actions$: Actions,
+    private authService: AuthService,
+    private snackbarService: SnackbarService,
+    private router: Router,
+  ) {}
+
+  Login$: Observable<Action> = createEffect(() => {
```
Comment (AleaJactaEst) [inline src/app/auth/state/effects.ts:27]: i guess should be login$ isn't?

File: src/app/auth/state/effects.ts [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    return this.actions$.pipe(
+      ofType(login),
+      switchMap((action: ReturnType<typeof login>) =>
+        from(this.authService.login(action.email, action.password)).pipe(
+          map((loginResponseData: ILoginResponseData) => {
+            this.snackbarService.openSnackBar('Login Success');
+            localStorage.setItem(
```
Comment (AleaJactaEst) [inline src/app/auth/state/effects.ts:34]: let's create service to work with local storage

File: src/app/auth/state/effects.ts [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    private router: Router,
+  ) {}
+
+  Login$: Observable<Action> = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(login),
+      switchMap((action: ReturnType<typeof login>) =>
+        from(this.authService.login(action.email, action.password)).pipe(
+          map((loginResponseData: ILoginResponseData) => {
+            this.snackbarService.openSnackBar('Login Success');
```
Comment (AleaJactaEst) [inline src/app/auth/state/effects.ts:33]: 'Login Success' - not quite good text - please use from some popular apps, or ask English teacher

File: src/app/auth/state/effects.ts [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    private authService: AuthService,
+    private snackbarService: SnackbarService,
+    private router: Router,
+  ) {}
+
+  Login$: Observable<Action> = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(login),
+      switchMap((action: ReturnType<typeof login>) =>
+        from(this.authService.login(action.email, action.password)).pipe(
```
Comment (AleaJactaEst) [inline src/app/auth/state/effects.ts:31]: why do we nee from?

File: src/app/auth/state/reducer.ts [hunk 13]
```diff
@@ -0,0 +1,64 @@
+import {
+  createFeatureSelector,
+  createReducer,
+  createSelector,
+  on,
+} from '@ngrx/store';
+
+import {
```
Comment (AleaJactaEst) [inline src/app/auth/state/reducer.ts:8]: we cna use import * as LoginAction from..
LoginAction.smth

File: src/app/auth/state/reducer.ts [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    isRequesting: false,
+  })),
+);
+
+const selectAuthFeature = createFeatureSelector<IAuthState>('auth');
+export const selectRequestingStatus = createSelector(
+  selectAuthFeature,
+  (state: IAuthState) => state.isRequesting,
+);
+export const selectUserId = createSelector(
```
Comment (AleaJactaEst) [inline src/app/auth/state/reducer.ts:61]: selectors should be in separate file

File: angular.json [hunk 15]
```diff
@@ -1,118 +1,109 @@
...
+          "options": {
+            "lintFilePatterns": ["src/**/*.ts", "src/**/*.html"]
+          }
+        }
+      }
+    }
+  },
+  "cli": {
+    "schematicCollections": ["@angular-eslint/schematics"],
+    "analytics": "377e5ac6-d9e6-4bec-89a5-b80db1342564"
```
Comment (Anastasiia-Svechnikova) [inline angular.json:107]: analytics is probably not neccessary

File: src/app/auth/components/auth/auth.component.html [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+          mat-raised-button
+          type="submit"
+          color="primary"
+          [disabled]="!form.valid">
+          {{ title }}
+        </button>
+      </form>
+      <div class="links">
+        <a
+          routerLink="/register"
```
Comment (Anastasiia-Svechnikova) [inline src/app/auth/components/auth/auth.component.html:50]: let's better put structural directive first (*ngif)

File: src/app/auth/services/auth.service.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { ILoginResponseData } from '../models/login-response-data';
+import { environment } from '../../../environments/environment';
```
Comment (Anastasiia-Svechnikova) [inline src/app/auth/services/auth.service.ts:6]: It would be better with full path

File: src/app/auth/state/effects.ts [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+    private localService: LocalService,
+  ) {}
+
+  login$: Observable<Action> = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(AuthActions.login),
+      switchMap((action: ReturnType<typeof AuthActions.login>) =>
+        this.authService.login(action.credentials).pipe(
+          concatMap((loginResponseData: ILoginResponseData) => {
+            this.snackbarService.openSnackBar('Login Successfully');
```
Comment (Anastasiia-Svechnikova) [inline src/app/auth/state/effects.ts:30]: we can write smth like 'Welcome to I Team! You can start with editing your profile.'

**Review by Anastasiia-Svechnikova — CHANGES_REQUESTED** [review]

**Review by Anastasiia-Svechnikova — APPROVED** [review]

File: src/app/auth/state/effects.ts [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+    private router: Router,
+    private localService: LocalService,
+  ) {}
+
+  login$: Observable<Action> = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(AuthActions.login),
+      switchMap((action: ReturnType<typeof AuthActions.login>) =>
+        this.authService.login(action.credentials).pipe(
+          concatMap((loginResponseData: ILoginResponseData) => {
```
Comment (AleaJactaEst) [inline src/app/auth/state/effects.ts:29]: why do we need ** concatMap** here?

File: src/app/auth/state/reducer.ts [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+    ...state,
+    isRequesting: false,
+  })),
+
+  on(AuthActions.register, (state) => ({
+    ...state,
+    isRequesting: true,
+  })),
+
+  on(AuthActions.registerSuccess, (state) => ({
```
Comment (AleaJactaEst) [inline src/app/auth/state/reducer.ts:36]: for same function handlere we can combine on methods

File: src/app/auth/state/selectors.ts [hunk 21]
```diff
@@ -0,0 +1,9 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+
+import { IAuthState } from './reducer';
+
+const selectAuthFeature = createFeatureSelector<IAuthState>('auth');
```
Comment (AleaJactaEst) [inline src/app/auth/state/selectors.ts:5]: empty line please

File: src/app/shared/services/local.service.ts [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+  providedIn: 'root',
+})
+export class LocalService {
+  saveData(key: string, value: string): void {
+    localStorage.setItem(key, value);
+  }
+
+  getData(key: string): string | null {
+    return localStorage.getItem(key);
+  }
```
Comment (AleaJactaEst) [inline src/app/shared/services/local.service.ts:13]: empty line please

**Review by Anastasiia-Svechnikova — APPROVED** [review]

**Review by AleaJactaEst — APPROVED** [review]

**Pull request merged**